docs/book/src/developer/providers/contracts/infra-cluster.md

```diff
@@ -0,0 +1,516 @@
+
+The goal of an InfraCluster resource is to supplie whatever prerequisites (in term of infrastructure) are necessary for running machines.
+Examples might include networking, load balancers, firewall rules, and so on.
+
+The InfraCluster resource will be referenced by one of the Cluster API core resources, Cluster.
+
+The [Cluster's controller](../../core/controllers/cluster.md) will be responsible to coordinate operations of the InfraCluster,
+and the interaction between the Cluster's controller and the InfraCluster resource is based on the contract
+rules defined in this page.
+
+Once contract rules are satisfied by an InfraCluster implementation, other implementation details 
+could be addressed according to the specific needs (Cluster API in not prescriptive).
+
+Nevertheless, it is always recommended to take a look at Cluster API controllers,
+in tree providers, other providers and use them as a reference implementation (unless custom solutions are required 
+in order to address very specific needs).
+
+In order to facilitate the initial design for each InfraCluster resource, a few [implementation best practices] and [infrastructure Provider Security Guidance]
+are explicitly called out in dedicated pages.
+
+<aside class="note warning">
+
+<h1>Never rely on Cluster API behaviours not defined as a contract rule!</h1>
+
+When developing a provider, you MUST consider any Cluster API behaviour that is not defined by a contract rule
+as a Cluster API internal implementation detail, subject to change without any advice.
+
+When implementing an Infra Cluster, you MUST NOT rely on any Cluster API internal behaviour, otherwise you are exposing users
+to the risk that such internal behaviour will change.
+
+Instead, whenever you need something more from the Cluster API contract, you MUST engage the community.
+
+The Cluster API maintainers welcome feedback and contributions to the contract in order to improve how it's defined,
+its clarity and visibility to provider implementers and its suitability across the different kinds of Cluster API providers.
+
+To provide feedback or open a discussion about the provider contract please [open an issue on the Cluster API](https://github.com/kubernetes-sigs/cluster-api/issues/new?assignees=&labels=&template=feature_request.md)
+repo or add an item to the agenda in the [Cluster API community meeting](https://git.k8s.io/community/sig-cluster-lifecycle/README.md#cluster-api).
+
+</aside>
+
+## Rules (contract version v1beta1)
+
+| Rule                                                                      | Mandatory | Note                                                                |
+|---------------------------------------------------------------------------|-----------|---------------------------------------------------------------------|
+| [All resources scope]                                                     | Yes       |                                                                     |
```

Comment:
Some character to highlight the first part would make this easier to read, e.g.

[All resources] scope

Or

All resources: scope

(I prefer the first option)
(Same for the headlines below of course)

Reply:
[] exists because those are links 😅 
Accordingly, option 1 is not viable, implemented option 2